### PR TITLE
add(catalog-import): Add some end-user Catalog import instructions

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Today I walked one of my colleagues through the register component button for the first time, we realized we both got a bit confused. He by the new concept, and me by the recent changes adding the new (and very cool) pull request feature and what it really meant to be an Entity File URL. These instructions are a bit wordy compared to what was there, but may help newer users understand how the registration works a bit more.

Also cleaned up the text on the support button and proper cased Backstage in the subtitle.

I figure if/when this is later enhanced to support GitHub, Azure, etc, the wording can be tweaked.

Happy to take feedback/adjust !

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by stefanalund — APPROVED** [review]

**Comment by stefanalund** [comment]
Side note: I would open a ticket about the browser scroll bar being visible on the left 😬

**Comment by stefanalund** [comment]
Another option for this information is to create a second column inside a InfoCard/Grid right of the stepper. There is already a lot of screen real estate not being used. In that case I would also make the different alternatives a numbered list to improve readability. 

What do you think @adamdmharvey ?

**Comment by adamdmharvey** [comment]
I like that idea, @stefanalund ; I'll give it a shot.

**Comment by adamdmharvey** [comment]
Latest

image

File: plugins/catalog-import/src/components/ImportComponentPage.tsx [hunk 1]
```diff
@@ -57,14 +57,44 @@ export const ImportComponentPage = ({
...
-            Start tracking your component in Backstage. TODO: Add more
-            information about what this is.
+            Start tracking your component in Backstage by adding it to the
+            software catalog.
           </SupportButton>
         </ContentHeader>
-        <Grid container spacing={3} direction="column">
-          <Grid item>
+        <Grid container spacing={3} direction="row-reverse">
+          <Grid item xs={5}>
```
Comment (stefanalund) [inline plugins/catalog-import/src/components/ImportComponentPage.tsx:67]: I thought these needed to be divisible by 12? I’d suggest changing to 4+8 on big screens and maybe even 12 (only one column) on the smallest screen sizes.

**Review by stefanalund — COMMENTED** [review]
Definitely prefer this way, nice work!

File: plugins/catalog-import/src/components/ImportComponentPage.tsx (hunk 1 above)
Comment (adamdmharvey) [reply plugins/catalog-import/src/components/ImportComponentPage.tsx:67]: According to https://material-ui.com/components/grid/, it's broken up into 12, but doesn't necessarily need to be divisible by 12.  In the sense that, there are always 12 to a grid, and any numbers that added up for columns that = 12 will be shown on a row?  That's how I understand it. Since there was still whitespace in the input box card, that's why I made it 8 and 5.  Can for sure be tweaked.

But I'll give a few reviews of what it looks like on mobile using Edge(Chrome) device emulation and throw a tweak or two in.

File: plugins/catalog-import/src/components/ImportComponentPage.tsx [hunk 2]
```diff
@@ -57,14 +57,44 @@ export const ImportComponentPage = ({
...
+                title: 'Learn more about the Software Catalog',
+                link:
+                  'https://backstage.io/docs/features/software-catalog/software-catalog-overview',
+              }}
+            >
+              <Typography variant="body2" paragraph>
+                There are two ways to register an existing component.
+                <ol>
+                  <li>
+                    <b>GitHub Repo</b>
```
[the remark this answers is not in the document]
Comment (adamdmharvey) [reply plugins/catalog-import/src/components/ImportComponentPage.tsx:79]: Sure. My experience with React is novice, at best, so I think there are some things I can do like `<Text style="bold">` (roughly), but not sure how to do numbered lists. I'll check it out! Welcome any advice from others here.

File: plugins/catalog-import/src/components/ImportComponentPage.tsx (hunk 1 above)
Comment (freben) [reply plugins/catalog-import/src/components/ImportComponentPage.tsx:67]: Yep you understood it right, it's just that we have never used 5. 3, 4, and 6 (OK, and 12) are used in the codebase because they evenly divide things in halves/thirds/quarters.

File: plugins/catalog-import/src/components/ImportComponentPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefanalund) [reply plugins/catalog-import/src/components/ImportComponentPage.tsx:79]: Look at `<Typography variant="xyz" paragraph>` and how it is used across the code base. If you need to use custom styling, please have a look at this example: https://github.com/backstage/backstage/blob/master/packages/core/src/components/WarningPanel/WarningPanel.tsx#L33

[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog-import/src/components/ImportComponentPage.tsx:79]: As for the list itself, i think `<ol><li>` is fine. But do use Typography for the text blocks inside.

(i also responded in Discord about this same thing)

**Comment by adamdmharvey** [comment]
Updated using even cards (50%/50% with 6/6=12 agrid) and pure Typography with no raw HTML.  It's not great on mobile (the input card gets weirdly wrapped as opposed to putting one on top of the other?), but OK for a swag I think?

image

iPhone X:
image

**Review by freben — APPROVED** [review]
Nice! Merge and tweeeeeak (if necessary)!

File: plugins/catalog-import/src/components/ImportComponentPage.tsx [hunk 3]
```diff
@@ -57,14 +57,41 @@ export const ImportComponentPage = ({
...
-            Start tracking your component in Backstage. TODO: Add more
-            information about what this is.
+            Start tracking your component in Backstage by adding it to the
+            software catalog.
           </SupportButton>
         </ContentHeader>
-        <Grid container spacing={3} direction="column">
-          <Grid item>
+        <Grid container spacing={3} direction="row-reverse">
+          <Grid item xs={6}>
```
Comment (freben) [inline plugins/catalog-import/src/components/ImportComponentPage.tsx:67]: @adamdmharvey If you want to tweak at a later point:

xs is just one of the screen sizes; there's md, lg, xl etc. You can specify multiple for different ranges. You may want to experiment so that

- both left and right are 6 at lg
- it's a 8 + 4 division at md
- it's a 12 + 12 division at xs (which also covers sm) which means it'll be two columns

or something like that!
